In [10]:
import os
import sys
import numpy as np
import pylab as pl


import json
import os

%pylab inline


Populating the interactive namespace from numpy and matplotlib


In [11]:
alpha=0.05
# we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0 = 3.5*0.01 
P_1 = 70.1*0.01

if P_0-P_1 >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 564

#lets get the counts by multiplying by the sample size
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

we must assess the statistical significance


In [12]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.421047276465


In [13]:
# I am goonna create a little one line function to calculate the standard dev, 
# it is not really needed, but just to show you how you do such a thing
sp_stdev = lambda p, n: np.sqrt( p * ( 1 - p ) / n[0] +  p * ( 1 - p ) / n[1]  )


sp_stdev_2y = sp_stdev(( Nt_0 + Nt_1) / (n_0 + n_1), [n_0, n_1])
print (P_0, n_0, n_1, sp_stdev_2y)

(0.035, 409, 564, 0.03206580860573819)


In [15]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print (z_2y)

20.7697865408


In [16]:
p_2y = 1 - 0.9998


def report_result(p,a):
    print ('is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


# Null hypothesis:
 the % of former prisoners recidivismed after release is the same or higher for candidates who participated in the program than for the control group, significance level p=0.05

Ho: P1 > = P0

H1: P1 < P0

In [17]:
alpha=0.05
# we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0 = 11.7*0.01 
P_1 = 10.0*0.01

if P_1 - P_0 >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 568

#lets get the counts by multiplying by the sample size
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

we must assess the statistical significance


In [18]:
chisqstat= lambda N, values, expect_num : ((values[0][0] * values[1][1]
                                            -values[0][1] * values[1][0])**2) * N / expect_num
Ntot = n_0 + n_1 # a + b + c + d = tot
tot_felony = P_0 * n_0 + P_1 * n_1
tot_non_felony = (1-P_0) * n_0 + (1-P_1) * n_1

expected_num = 568 * 409 * tot_felony * tot_non_felony
sample_values = [[P_1 * n_1, (1-P_1) * n_1], [P_0 * n_0, (1-P_0) * n_0]]
 

print (chisqstat(Ntot,  sample_values, expected_num))

0.718493917505


| convicted of a fellony | convicted of a felony | non convicted of a felony |
|:------------:|:----------:|:-----------:|
| test sample |56.8|511.2 |
| control sample |40.9|361.1| 
| total | 97.7|872.3|

# DOF = Number of observations - number of Independent Variables¶ where here DOF = 1

# the chi-squared statistics we got from the test is 0.718493917505, which is less than 3.84 (from table above, degrees of freedom 1 and alpha = 0.05). We can't conclude that the test rejects H0.

Alpha = 0.05 defines the term 'statistically impossible'.